public/src/components/teacher/Results.tsx

```diff
@@ -143,6 +143,23 @@ export class Results extends React.Component<IResultsProps, IResultsState> {
             const ans = await this.props.onSubmissionStatusUpdate(selected);
             if (ans) {
                 selected.approvedDate = new Date().toLocaleString();
+
+                if (current && selected.groupid > 0) {
```

Comment:
A few things in this func. As a random reviewer, it is a bit hard to follow. But it would help if `current` was named `currentSubmission` and `selected` was named `selectedSubmission`. And maybe adding a few comments (not many)... to explain what's going on.

Reply:
Noted and updated.